lib/asset_writer.rb

```diff
@@ -0,0 +1,37 @@
+class AssetWriter
+  attr_reader :asset, :target_path
+
+  delegate :logical_path, :digest_path, :to => :asset
+
+  def initialize(asset_or_path, target_dir)
+    @asset = find_asset(asset_or_path)
+    @target_dir = target_dir
+    @target_path = Rails.root.join(target_dir, @asset.digest_path)
```

Comment:
That is, I'm wondering if this code should just deal with paths only, and the caller needs to pass digest_path or something.